modules/platforms/cpp/ignite/schema/tuple_assembler.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <ignite/schema/binary_tuple_schema.h>
+#include <ignite/schema/binary_tuple_builder.h>
+
+#include <cstring>
+#include <optional>
+#include <tuple>
+#include <type_traits>
+#include <vector>
+
+/**
+* This used to be the main tool to construct binary tuples (in IEP-54 format).
+* Now this is just a helper for unit tests.
+*/
+class tuple_assembler {
+private:
+    ignite::binary_tuple_schema schema;
+
+    ignite::binary_tuple_builder builder;
+
+public:
+    /** Construct a new Tuple Assembler object. */
+    explicit tuple_assembler(ignite::binary_tuple_schema &sch)
+        : schema(std::move(sch))
+        , builder(schema.num_elements()) { }
+
+    /** Start a new tuple. */
+    void start() {
+        builder.start();
+    }
+
+    /** Append a null value. */
+    void appendNull() {
```

Comment:
1. `append_null` 2. is it still used somewhere?

Reply:
Only in a currently muted test

Reply:
It's not clear how it is going to work now.

Reply:
Added an append